### ISSUE TITLE
Endless reconnects

### ISSUE THREAD

**Issue by Brandantl** [yourWaifu/sleepy-discord#115]
I don't know whos bug this is but ill mention it here for the time being. After 8 days of uptime I encountered an issue where the bot tried to connect to Discord 1000 times in a short amount of time. Calling onReady a 1000 times, thus Discord revoked my token. 

![image](https://user-images.githubusercontent.com/6909044/45002921-36d75300-af91-11e8-9e6d-c412c516714d.png)

**Comment by yourWaifu** [comment]
I'm aware of this issue. It has happened to be me a few times already. I don't exactly know why it's happening and I've been making changes that I think would fix it. However, there's currently no way for me to know if it's fixed or not without having to wait at least a week for it to happen.

**Comment by Brandantl** [comment]
On Visual Studio you could try a hit counter breakpoint that breaks after like 500 reconnects.

**Comment by yourWaifu** [comment]
The bot that use that runs for a week is using Ubuntu. I can use GDB to do it, but that would still require me to wait for it happen.

**Comment by Moulijin** [comment]
![image](https://user-images.githubusercontent.com/2942447/45580941-cd3a2d00-b84b-11e8-9a39-66cdf58dc3bc.png)
Is this it? lol

**Comment by yourWaifu** [comment]
yea that looks like the issue

**Comment by Brandantl** [comment]
I think it happened to everyone who uses my software and your library at the exact same time which as 13:00 today.

**Comment by Brandantl** [comment]
This was my method to detect it but then I figured out everytime it reconnects it doesn't invoke onReady
![image](https://user-images.githubusercontent.com/6909044/46238357-24f29100-c340-11e8-8f5c-e39571325bb3.png)

Know any function that'll be invoked on each reconnect?

**Comment by yourWaifu** [comment]
Yes, onDisconnect should be called or invoked on every reconnect.

**Comment by yourWaifu** [comment]
I haven't found the root cause, but for now, I'll make it delay between reconnects by increasing the time til reconnecting again on each fail and placing a limit on the number of times it can automatically fail reconnecting in a row.
